src/app/modules/submission/email-submission/__tests__/email-submission.routes.spec.ts

```diff
@@ -44,21 +44,20 @@ jest.mock('nodemailer', () => ({
 }))
 
 jest.mock('@opengovsg/myinfo-gov-client', () => ({
-  MyInfoGovClient: jest.fn(),
+  MyInfoGovClient: jest.fn().mockReturnValue({
+    extractUinFin: jest.fn(),
+  }),
   MyInfoMode: jest.requireActual('@opengovsg/myinfo-gov-client').MyInfoMode,
   MyInfoSource: jest.requireActual('@opengovsg/myinfo-gov-client').MyInfoSource,
   MyInfoAddressType: jest.requireActual('@opengovsg/myinfo-gov-client')
     .MyInfoAddressType,
   MyInfoAttribute: jest.requireActual('@opengovsg/myinfo-gov-client')
     .MyInfoAttribute,
 }))
-const MockMyInfoGovClient = mocked(MyInfoGovClient, true)
-const mockExtractUinFin = jest.fn()
-MockMyInfoGovClient.mockImplementation(
-  () =>
-    (({
-      extractUinFin: mockExtractUinFin,
-    } as unknown) as MyInfoGovClient),
+
+const MockMyInfoGovClient = mocked(
+  new MyInfoClient.MyInfoGovClient({} as IMyInfoConfig),
```

Comment:
wow how does this not throw an error? myinfogovclient definitely throws if params are missing

Reply:
i think this is allowed because we call `jest.mock` on the containing module. this transforms MyInfoGovClient into a mock function, whose return value is then set later on!

Reply:
ah so you're calling `new` on the mocked module. got it